### ISSUE TITLE
Excessive POST {storageAccount}/Tables() from [Table] attribute binding

### ISSUE THREAD

**Issue by marcinzi** [Azure/azure-webjobs-sdk#2322]
We noticed on telemetry a lot of 409s from table storage dependency.
POST https://{our account}.table.core.windows.net:443/Tables()
result code 409

Looking for root cause found that [Table] attribute on binding to CloudTable object is most likely responsible for ensuring that table exists. 

#### Repro steps

Run Azure Function bound to CloudTable parameter

#### Expected behavior

No excessive POST call is done. We're in control of our Azure Dependencies and binding a parameter should not try to modify our Azure Resources.
At the minimum this behavior should be parametrized, and default off.

#### Actual behavior

Every AF invocation causes the POST call,
this is performance overhead

#### Known workarounds

Not known

#### Related information 

Most likely this line of code:
https://github.com/Azure/azure-webjobs-sdk/blob/ef265e20d4357f2a82871b63a778c99cb6aac8db/src/Microsoft.Azure.WebJobs.Extensions.Storage/Tables/Config/TablesExtensionConfigProvider.cs#L216

**Comment by brettsam** [comment]
Yes, that seems aggressive. We can change this to check for existence first.

**Comment by yarseyah** [comment]
Could this not be set to never create by the user of binding, e.g. with an input parameter of "ThrowIfNotExists" on the binding? 

The existence of the tables is quite fundamental in my use-case, so getting 500 errors from function would be preferable to a 409 in every call.

**Comment by metu** [comment]
Our team is facing this issue at the moment. We have a .net core 3 web API which receives 100s of requests per second and we've been trying to solve socket exhaustion issues and the like for a while now (we're calling other APIs etc). We've gotten quite far but have hit a wall on this issue. In App Insights it shows that dependency calls can take 50 - 90 seconds and fails with the 409 error code. @marcinzi thanks for pointing out the line of code causing the issue. We've removed the line and have taken @qngEdg 's advice and only inject CloudTable when needed. Deployments are pending... will update once I've tested all this.
